core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -274,9 +279,28 @@ abstract class BaseConsumerTest extends IntegrationTestHarness with Logging {
   }
 
   protected def sendRecords(numRecords: Int, tp: TopicPartition) {
-    (0 until numRecords).map { i =>
-      this.producers(0).send(new ProducerRecord(tp.topic(), tp.partition(), s"key $i".getBytes, s"value $i".getBytes))
-    }.foreach(_.get)
+    sendRecords(numRecords, tp, None)
+  }
+
+  protected def sendRecords(numRecords: Int, tp: TopicPartition, codec: Option[String] = None) {
```

Comment:
After a second look I'm not a big fan of this change. Basically, the `IntegrationTestHarness` is ill-equipped to handle multiple types of producers. These changes make the API a bit misleading. For e.g., if the producer-config that you supply to the harness is configured with compression turned on then the preceding `sendRecords` API would send compressed records and this would be redundant. While this sounds out of scope for this jira I'm wondering if it is worth thinking of a simple way to improve this.